### ISSUE TITLE
Unable to instrument to trace custom spawning child process background scripts

### ISSUE THREAD

**Issue by johnson-imaginea** [appneta/ruby-traceview#176]
Following is the piece of code I wanted to enable tracing the background process scripts. I tried using `TraceView::API.start_trace` & `TraceView::API.trace` methods but its not sending the traces to appneta dashboard and I'm getting the error **Send failed - reporter returned -1 [oboe.c:883] [Oboe-lib-error p#17895]**
```ruby
def background_job
  TraceView::API.start_trace('background Deamon') do
   Process.fork do
      Daemons.daemonize(DAEMONIZATION_OPTIONS.merge(app_name: 'job'))
      ActiveRecord::Base.establish_connection
      set_process_title('Starting')
      create_spawned_log
      trap_exit
      setup_environment
      TraceView::API.trace('job') do
        run!
      end
    end
 end
end
```

Please help me to get it work. Thanks in advance.

**Comment by pglombardo** [comment]
Hi @johnson-imaginea - I see two issues:

1. the tracing won't work if you attempt to trace across a process fork

Instead, start the trace (`start_trace`) after the fork and also call `TraceView::Reporter.restart` (similar to the `ActiveRecord::Base.establish_connection` call) to reinitialize the reporter.

2. Daemons can run for days, months etc and this trace would never finish to be reported

Instead you could trace _events_ in your Daemon and wrap that activity with `start_trace`.


```ruby
def background_job
    Process.fork do
      Daemons.daemonize(DAEMONIZATION_OPTIONS.merge(app_name: 'job'))
      ActiveRecord::Base.establish_connection
      TraceView::Reporter.restart
      set_process_title('Starting')
      create_spawned_log
      trap_exit
      setup_environment
      run!
  end
end

# Example daemon event handling code
# (I used video encoding but could be anything)

def daemon_handle_some_video_encoding
  TraceView::API.start_trace('video_upload') do
    # work, work work

    TraceView::API.trace('video_identify') do
      # identify work
    end

    TraceView::API.trace('video_encode') do
      # video encoding work
    end
  end
end
```

Hopefully this helps.  Any questions, please let me know!

**Comment by johnson-imaginea** [comment]
Hi @pglombardo 

Thank you very much for the support and for the reference codes. I just followed your recommendation, it works now successfully.

But **TraceView::Reporter.restart** didn't work, it threw error undefined method, so I replaced it by the reporter re initialization code as suggested in the appneta documentation **TraceView.reporter = TraceView::UdpReporter.new(TraceView::Config[:reporter_host], TraceView::Config[:reporter_port])**

Once again I appreciate your help and thanks lot.
